src/combat.cpp

```diff
@@ -307,7 +308,7 @@ ReturnValue Combat::canDoCombat(Creature* attacker, Creature* target)
 			}
 
 			// nopvp-zone
-			const Tile* targetPlayerTile = targetPlayer->getTile();
+			const std::shared_ptr<Tile> targetPlayerTile = targetPlayer->getTile();
```

Comment:
Could this be `const auto&`?
```suggestion
			const auto& targetPlayerTile = targetPlayer->getTile();
```